tests/e2e/tests/tests.spec.ts

```diff
@@ -0,0 +1,170 @@
+import MapComponentInfoPopover from './MapComponent/MapComponentInfoPopover';
+import MapComponentInteractive from './MapComponent/MapComponentInteractive';
+import MapComponentLayersDefault from './MapComponent/MapComponentLayersDefault';
+import MapComponentLayersVisible from './MapComponent/MapComponentLayersVisible';
+import MapComponentTriggerLayer from './MapComponent/MapComponentTriggerLayer';
+import MapComponentVisible from './MapComponent/MapComponentVisible';
+import TimelineComponentDisabled from './TimelineComponent/TimelineComponentDisabled';
+import TimelineComponentNotClickable from './TimelineComponent/TimelineComponentNotClickable';
+import TimelineComponentVisible from './TimelineComponent/TimelineComponentVisible';
+import UserStateComponentLogout from './UserStateComponent/UserStateComponentLogout';
+import UserStateComponentVisible from './UserStateComponent/UserStateComponentVisible';
+
+const datasets: Dataset[] = [UgandaFloodsNoTrigger, UgandaFloodsTrigger];
+
+datasets.forEach((dataset) => {
+  const {
+    country: { code },
+    hazard,
+    scenario,
+    user: { email, password },
+  } = dataset;
+
+  let accessToken: string;
+  let page: Page;
+
+  const pages: Partial<Pages> = {};
+  const components: Partial<Components> = {};
+
+  test.describe(`Dataset: ${email} ${code} ${hazard} ${scenario}`, () => {
+    const date = new Date();
+
+    test.beforeAll(async ({ browser }) => {
+      page = await browser.newPage();
+      // Initialize pages and components after sharedPage is assigned
+      pages.login = new LoginPage(page);
+      pages.dashboard = new DashboardPage(page);
+      components.map = new MapComponent(page);
+      components.userState = new UserStateComponent(page);
+      components.aggregates = new AggregatesComponent(page);
+      components.chat = new ChatComponent(page);
+      components.disasterType = new DisasterTypeComponent(page);
+      components.timeline = new TimelineComponent(page);
+
+      // Reset the database
+      accessToken = await getAccessToken();
+      await resetDB(accessToken);
+
+      // Load a mock scenario
+      await mockData(hazard, scenario, code, accessToken, date);
+
+      await page.goto('/');
+      // Login into the portal
+      await pages.login.login(email, password);
+    });
+
+    test.beforeEach(async ({ page }) => {
+      await page.goto('/');
+    });
+
+    test.afterAll(async () => {
+      await page.close();
+    });
+
+    test.describe('DashboardPage', () => {
+      DashboardPageVisible(pages, components, dataset, date);
+    });
+
+    test.describe.only('MapComponent', () => {
+      MapComponentVisible(pages, components, dataset);
+      MapComponentInteractive(pages, components, dataset);
+      MapComponentInfoPopover(pages, components, dataset);
+      MapComponentLayersVisible(pages, components, dataset);
+      MapComponentTriggerLayer(pages, components, dataset);
+      MapComponentGloFASStations(pages, components, dataset);
```

Comment:
I see this repeats a lot of tests across both no-trigger and trigger, unlike before. I get that may be intentional, but it comes at the cost of more running time / more flakiness opportunity. Is that worth it? There are now 47 tests run (vs 29 before I think).

Reply:
accepted for now